docarray/doc_index/backends/weaviate_doc_index.py

```diff
@@ -0,0 +1,594 @@
+                if column_info.db_type not in WEAVIATE_PY_VEC_TYPES:
+                    raise ValueError(
+                        f'Column {column_name} is marked as embedding but is not of type {WEAVIATE_PY_VEC_TYPES}'
+                    )
+
+        if num_embedding_columns > 1:
+            raise ValueError(
+                f'Only one column can be marked as embedding but found {num_embedding_columns} columns marked as embedding'
+            )
+
+    def _set_embedding_column(self) -> None:
+        for column_name, column_info in self._column_infos.items():
+            if column_info.config.get('is_embedding', False):
+                self.embedding_column = column_name
+                break
+
+    def _configure_client(self) -> None:
+        self._client.batch.configure(**self._runtime_config.batch_config)
+
+    def _create_schema(self) -> None:
+        schema = {}
+
+        properties = []
+        column_infos = self._column_infos
+
+        for column_name, column_info in column_infos.items():
+            # in weaviate, we do not create a property for the doc's embeddings
+            if column_name == self.embedding_column:
+                continue
+            prop = {
+                "name": column_name
+                if column_name != 'id'
+                else DOCUMENTID,  # in weaviate, id and _id is a reserved keyword
+                "dataType": column_info.config["dataType"],
+            }
+            properties.append(prop)
+
+        # TODO: What is the best way to specify other config that is part of schema?
+        # e.g. invertedIndexConfig, shardingConfig, moduleConfig, vectorIndexConfig
+        #       and configure replication
+        # we will update base on user feedback
+        schema["properties"] = properties
+        schema["class"] = self._db_config.index_name
+
+        # TODO: Use exists() instead of contains() when available
+        #       see https://github.com/weaviate/weaviate-python-client/issues/232
+        if self._client.schema.contains(schema):
+            logging.warning(
+                f"Found index {self._db_config.index_name} with schema {schema}. Will reuse existing schema."
+            )
+        else:
+            self._client.schema.create_class(schema)
+
+    @dataclass
+    class DBConfig(BaseDocumentIndex.DBConfig):
+        host: str = 'http://weaviate:8080'
+        index_name: str = 'Document'
+
+    @dataclass
+    class RuntimeConfig(BaseDocumentIndex.RuntimeConfig):
+        default_column_config: Dict[Type, Dict[str, Any]] = field(
+            default_factory=lambda: {
+                np.ndarray: {
+                    'dataType': ['number[]'],
+                },
+                docarray.typing.ID: {'dataType': ['string']},
+                bool: {'dataType': ['boolean']},
+                int: {'dataType': ['int']},
+                float: {'dataType': ['number']},
+                str: {'dataType': ['text']},
```

Comment:
This dataType information should not be encoded here, rather it should be in the python_type_to_db_type method.
There are two reasons for this:
1) if done that way, and if the user manually wants to specify a different `col_type` in the `Field(..)` notation, the abstract class can automatically use the user defined type instead of the default one. This will then be reflected in the corresponding _ColumnInfo, and you don't have to worry about it
2) it allows you to use `issubclass()` etc to handle multiple classes and subclasses etc at once, so less writing for you :)

You can take a look at the ES PR and how it is done there